### PR TITLE
Upload files to create a new project

### PR DESCRIPTION
**Pull request by iannbing**
## Description

This PR makes it possible to create a project by uploading files/folders from user's computer.

Fixes: #870 

## Checklist
- [x] The PR branch is set up to merge into `dev` instead of `master`.

### CODE REVIEW

**Comment by ncspost** [comment]
Great work @iannbing 

# Some strange behaviours

I get a strange behaviour where it hangs on 98% when trying to upload a folder...
https://user-images.githubusercontent.com/6852479/166452862-9a3a88b6-6482-4ce5-aeb2-a123fd11ea68.mp4

I then refresh the page (because there's no cancel button) and get some strange values in the project list...
CleanShot 2022-05-03 at 13 30 39@2x

When git repo text input clicked and exited with no URL - error input text appears... this should only happen when there is text in the box...
CleanShot 2022-05-03 at 13 37 47@2x


# Proposed changes

## Project list page
- [ ] Restyle drag & drop incoming [to this](https://www.figma.com/file/9FIzlSj1Yg2papVja4EeIZ/GH%23870-Upload-local-folder-(Solution)?node-id=131%3A24658)

## Import project dialog
- [ ] Change git repo text field error text to "Not a valid HTTPS git repository URL"
- [ ] When clicking out of git repo text field and it is empty... do not display error message
- [ ] When hovering over drag & drop box change icon colour to same colour as border
- [ ] Reduce vertical spacing between icon and "drag & drop" text from 8px ➡ 0px
- [ ] Change drag & drop box border style to dash 4, gap 4

## Uploading project dialog
- [ ] Is it possible to align the top of this dialog and previous dialog so the transition is less jarring?
- [ ] Add "Cancel" text button next to "Save" button
- [ ] Change icon colour to light/action/active (54p)
- [ ] Change text size to "body 2"
- [ ] Change text colour to light/secondary/dark
- [ ] Will there actually ever be any Jobs shown from imported folder? If not, should we remove that icon?

## Upload complete dialog
- [ ] Is it possible to align the top of this dialog and previous dialog so the transition is less jarring?
- [ ] Change dialog heading to "Upload complete"
- [ ] When hitting save - open up new project rather than showing project list

**Comment by iannbing** [comment]
@ncspost 

> Restyle drag & drop incoming [to this](https://www.figma.com/file/9FIzlSj1Yg2papVja4EeIZ/GH%23870-Upload-local-folder-(Solution)?node-id=131%3A24658)

The current styling is aligned with File Manager. I'd rather change them altogether.

> Change drag & drop box border style to dash 4, gap 4

This requires some dirty hack. I'm not sure if it's worth it to achieve this.

> Is it possible to align the top of this dialog and previous dialog so the transition is less jarring?

Unfortunately MUI dialog doesn't offer an API to do this. I'd rather not to hack it because we might break its built-in responsiveness.

> When hitting save - open up new project rather than showing project list

It should have already be functioning as such (Once you hit "SAVE" it should navigate to PIpelineEditor and open the first pipeline if any.). Could you confirm it? Let me know if you found a bug.

**Comment by ncspost** [comment]
> This requires some dirty hack. I'm not sure if it's worth it to achieve this.

np!

> Unfortunately MUI dialog doesn't offer an API to do this. I'd rather not to hack it because we might break its built-in responsiveness.

np!

**Comment by ricklamers** [comment]
image

The # of pipelines indicator gets stuck for me. No error in network tab or console.

**Comment by iannbing** [comment]
> and the button makes it feel like you're already "done". The button should probably be named "Save".

This change was proposed earlier by @ncspost. I had similar concerns, but then I was not super against it ("View project" tells user what's coming, i.e. immediately redirect to the project). I might opt for "Save and go to project", but it's quite verbose.

@ricklamers @ncspost could we make a decision here?

File: services/orchest-webserver/client/src/projects-view/ImportDialog.tsx [hunk 1]
```diff
@@ -30,212 +53,687 @@ const getMappedErrorMessage = (key: CreateProjectError | string | null) => {
...
+  const saveProjectName = async () => {
+    if (!newProjectUuid || projectNameValidation.error) return;
+    setImportStatus("SAVING_PROJECT_NAME");
+    try {
+      await fetcher(`/async/projects/${newProjectUuid}`, {
+        method: "PUT",
+        headers: HEADER.JSON,
+        body: JSON.stringify({ name: projectName }),
+      });
+      await fetchProjects();
```
[the remark this answers is not in the document]
Comment (ricklamers) [reply services/orchest-webserver/client/src/projects-view/ImportDialog.tsx:390]: As discussed we'll pick this up later by uploading the project to a temporary location. For now this is fine :+1: 

Nicely done.

**Comment by ricklamers** [comment]
I think this is ready! Great job all (mainly @iannbing ofc)

**Pull request merged**